code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -117,40 +113,32 @@ rcd light flash thingy on matter drain
 	if(C)
 		C.post_status("alert", "lockdown")
 
-	src.verbs += /mob/living/silicon/ai/proc/disablelockdown
-	src << "<span class = 'warning'>Lockdown Initiated.</span>"
-	malf_cooldown = 1
-	spawn(30)
-	malf_cooldown = 0
+	verbs -= /mob/living/silicon/ai/proc/lockdown
+	minor_announcement.Announce("Hostile runtime detected in door controllers. Isolation lockdown protocols are now in effect. Please remain calm.","Network Alert")
+	src << "<span class='danger'>Lockdown initiated. Network reset in 90 seconds.</span>"
+	spawn(900) //90 Seconds.
+		disablelockdown() //Reset the lockdown after 90 seconds.

```

Comment:
If the AI is destroyed within 60 seconds of initiating a lockdown, it will be deleted by the GC and this spawned thread will terminate before `disablelockdown` gets called. To avoid that, you'd want to `spawn(-1)`, immediately null out `src`, sleep for 900, and then disable the lockdown, bearing in mind that you can't call the AI's `disablelockdown` because your `src` is gone.

... or it might be fairly safe to just assume that, if the AI is dead, the round is probably ending, and not worry about it. Depends on whether you want non-round-ending malfs to ever be properly supported.